Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Common.hs

```diff
@@ -616,5 +616,13 @@ buildAddressFromText fullAddress = do
       locality_ = Just $ splitedAddress List.!! 3
       state_ = Just $ splitedAddress List.!! 5
       street_ = Just $ splitedAddress List.!! 2
-      ward_ = Just $ T.intercalate ", " $ catMaybes [door_, building_, locality_]
-  pure $ OS.Address {area_code = area_code_, building = building_, city = city_, country = country_, door = door_, locality = locality_, state = state_, street = street_, ward = ward_}
+      door = replaceEmpty door_
+      building = replaceEmpty building_
+      locality = replaceEmpty locality_
+      ward_ = Just $ T.intercalate ", " $ catMaybes [door, building, locality]
+  if ward_ == Just ""
```

Comment:
you can have this `if` case only for populating `ward_` and need not duplicate the entire Address creation in if and else case